pkg/util/abiutil/abiutil.go

```diff
@@ -20,3 +27,56 @@ func MustParseABI(json string) abi.ABI {
 	}
 	return val
 }
+
+func ConvertType(in interface{}, proto interface{}) (out interface{}, err error) {
+	defer func() {
```

Comment:
have we encountered any panic?

Reply:
if it's there, chances are we will

Reply:
I don't we need it, if we do proper error checking, we should be good.
We have not encountered a case of a panic, at least to my knowledge, and no operator has reported one either

Reply:
@swarmHaseeb ^^^

Reply:
@istae well yes no one has reported any errors but if you look at test cases you can clearly see the edge cases that were originally missing, not today but tomorrow we can get an error. So this is proper error checking.

Reply:
so even the standard library packages have panics, and things panic when we write unrecoverable bugs and proper checking is not done

sticking a panic catcher here does not actually make the logic more "robust" it just sweeps potential bugs and problems with the code under the rug

Reply:
`ConvertType` function will convert potential panic to error, therefore all panicking problems in code will not be disregarded.

Code which returns error instead of panic is generally more robust because developer have to handle error, while panic have to be "discover". Later comes up with a price if panicking code is discovered in production. 

---

If `ConvertType` is blocking this PR, we could come to common grounds, in order to move with this PR forward, and remove this this function and use one from `abi` package.